extension/opampextension/config.go

```diff
@@ -0,0 +1,42 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opampextension
+
+import (
+	"errors"
+
+	"go.opentelemetry.io/collector/config"
+)
+
+// Config has the configuration for the opamp extension.
+type Config struct {
+	config.ExtensionSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
+
+	// Endpoint is the OpAMP server URL. Transport based on the scheme of the URL.
+	Endpoint string `mapstructure:"endpoint"`
```

Comment:
@tigrannajaryan I've experimented with https://github.com/SumoLogic/sumologic-otel-collector/blob/261043b945a4e69e73975cb3750f1b1d9f2c55c1/pkg/extension/opampextension/config.go#L26-L28

Reply:
Using `HTTPClientSettings` feels cumbersome and perhaps its going to be problematic. `StartSettings` (https://github.com/open-telemetry/opamp-go/blob/main/client/types/startsettings.go#L11) has me doing things like: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16594/commits/924ed2df770abeb0537015cb4b2102c6c65bb826. Also, most of the settings will go unused, which could be confusing since they will pass validation.